src/scos_tekrsa/hardware/tekrsa_sigan.py

```diff
@@ -71,6 +76,25 @@ def __init__(self):
 
         except Exception as error:
             logger.error(f"Unable to initialize sigan: {error}")
+            self.power_cycle_and_connect()
+
+    def power_cycle_and_connect(self, sleep_time=2.0):
```

Comment:
Should this be implemented at the `SignalAnalyzerInterface` level in SCOS Actions so it applies for all sigans? It seems to be essentially independent of the signal analyzer being used. If `SignalAnalyzerInterface.connect()` is implemented as an `@abstractmethod`, then `power_cycle_and_connect()` could be implemented at the class level in `SignalAnalyzerInterface`.

Currently, `scos-tekrsa`, `scos-usrp`, and even `scos-keysight` implement their own `connect()` methods, so creating `connect()` as an abstract method would not require any modification there.

The only way I see this not being a good approach is if a given signal analyzer implements its own API command which allows you to cycle the power. In that case, you might not want to call `scos_actions.hardware.utils.power_cycle_sigan()` to do the power cycle, and instead make a direct API call. But in that case, the specific signal analyzer's `SignalAnalyzerInterface` could override the class-level `power_cycle_and_connect()`.

Reply:
If this change is made, also remember to remove the imports for `power_cycle_sigan` and `HardwareConfigurationException` which would no longer be needed.

Reply:
Implemented in NTIA/scos-actions#44. Removing redundant methods from scos-tekrsa sigan interface in a moment.

Reply:
Done in [1010774](https://github.com/NTIA/scos-tekrsa/pull/26/commits/101077486f2107923d9dacb03de6ffca42c1259f). This PR now depends on the implementation of NTIA/scos-actions#44